backend/OpenAPI.yaml

```diff
@@ -137,37 +119,69 @@ paths:
           content:
             application/json:
               schema:
-                type: object
-                properties:
-                  error:
-                    type: string
-                    example: "Invalid listing ID format. Must be a valid ObjectId."
+                $ref: "#/components/schemas/ErrorResponse"
         "404":
           description: Listing not found
           content:
             application/json:
               schema:
-                type: object
-                properties:
-                  error:
-                    type: string
-                    example: "Listing not found."
+                $ref: "#/components/schemas/ErrorResponse"
+        "422":
+          description: Invalid Input body
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
         "500":
           description: Internal Server Error
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/500ErrorResponse"
+                $ref: "#/components/schemas/ErrorResponse"
 
 components:
   schemas:
-    500ErrorResponse:
+    ErrorResponse:
+      type: object
+      required:
+        - details
+      properties:
+        details:
+          type: string
+      example: 
```

Comment:
For this, the example is probably better left for the individual operations. So that you can give a better example error message that is specific to that endpoint and operation error.

Reply:
At the moment, this is what you get when you have the docs.

<img width="738" alt="image" src="https://github.com/user-attachments/assets/9111fbdc-ebe6-4470-9b56-82fd8856bffb" />

which is not good. In the interest of time, I'd say just remove this first. We can come up with better examples later. It's not crucial.